tests/test_selection/test_drop_high_psi_features.py

```diff
@@ -0,0 +1,543 @@
+
+    with pytest.raises(ValueError):
+        DropHighPSIFeatures(switch=1)
+
+    with pytest.raises(ValueError):
+        DropHighPSIFeatures(strategy="unknown")
+
+    with pytest.raises(ValueError):
+        DropHighPSIFeatures(min_pct_empty_bins="unknown")
+
+
+# ================= test fit() functionality ==================
+
+
+def test_split_col_not_included_in_variables(df):
+    """Check that the split columns is not included among the features
+    to evaluate when these are selected automatically."""
+    transformer = DropHighPSIFeatures(split_col="var_3", variables=None)
+    transformer.fit(df)
+
+    assert transformer.variables is None
+    assert "var_3" not in transformer.variables_
+    assert "var_3" not in transformer.psi_values_.keys()
+
+
+def test_missing_split_col(df):
+    """Test an error is raised if the split column contains missing values."""
+    data = df.copy()
+    data["var_3"].iloc[15] = np.nan
+
+    with pytest.raises(ValueError):
+        transformer = DropHighPSIFeatures(split_col="var_3")
+        transformer.fit_transform(data)
+
+
+def test_raise_error_missing_value_na(df):
+    """Test an error is raised when missing values is set to raise."""
+    data = df.copy()
+    data["var_3"].iloc[15] = np.nan
+
+    with pytest.raises(ValueError):
+        transformer = DropHighPSIFeatures(missing_values="raise")
+        transformer.fit(data)
+
+
+def test_missing_value_ignored(df):
+    """Test if PSI are computed when missing values are present in the dataframe."""
+    data = df.copy()
+    data["var_3"].iloc[15] = np.nan
+
+    var_col = [col for col in data if "var" in col]
+
+    transformer = DropHighPSIFeatures(missing_values="ignore")
+    transformed = transformer.fit_transform(data)
+
+    pd.testing.assert_frame_equal(transformed, data[var_col])
+
+
+def test_raise_missing_value_inf(df):
+    """Test an error is raised for inf when missing values is set to raise."""
+    data = df.copy()
+    data["var_3"].iloc[15] = np.inf
+
+    with pytest.raises(ValueError):
+        transformer = DropHighPSIFeatures(missing_values="raise")
+        transformer.fit(data)
+
+
+# ========= tests for _split_dataframe() fit ====
+
+
+quantile_test = [(0.5, 50), (0.33, 33), (0.17, 17), (0.81, 81)]
+
+
+@pytest.mark.parametrize("split_frac, expected", quantile_test)
+def test_calculation_quantile(split_frac, expected):
+    """Test the calculation of the quantiles using distinct values."""
+    df = pd.DataFrame(
+        {"A": [it for it in range(0, 101)], "B": [it for it in range(0, 101)]}
+    )
+
+    test = DropHighPSIFeatures(
+        split_col="A", split_frac=split_frac, split_distinct=False
+    )
+    test.fit_transform(df)
+    assert test.cut_off_ == expected
+
+
+def test_calculation_distinct_value():
+    """Test the calculation of the quantiles using distinct values."""
+    df = pd.DataFrame(
+        {"A": [it for it in range(0, 30)], "C": ["A", "B", "C", "D", "D", "D"] * 5}
```

Comment:
In: `"C": ["A", "B", "C", "D", "D", "D"] * 5` there are:

- 5 x A observations
- 5 x B observations
- 5 x C observations
- 15 x D observations

total = 30 observations

So if split_frac = 0.5 I expect the same result for both split_distinct=True or False ==> 50% is 15 labels, so that would be A, B and C in the basis set.

I think there is a problem with the logic in the class, or maybe I do not understand the logic.

A simpler test would be directly on  `"C": ["A", "B", "D", "D", "D", "D"] ` :

- if split_distinct=False I expect  ["A", "B", "D",] and ["D", "D", "D"]
- if split_distinct=True I expect  ["A", "B",,] and [ "D", "D", "D", "D"]

Also, can we  make sure split_distinct also works well when the split_col is numerical? That is:

If `"C": [1, 2, 3, 3, 3, 3] ` :

- if split_distinct=False I expect  [1,2,3,] and [3,3,3]
- if split_distinct=True I expect  1,2] and [ 3,3,3,3,]

Reply:
The behaviour is what I expect but also probably something I did not explained correctly. Split distinct will look at the vector of unique values and split that vector according to the split_frac. 
![image](https://user-images.githubusercontent.com/32060943/140817461-6b89aa69-645e-4877-913b-2051ce1493f5.png)

This can be used to split a dataframe according to the customer ID (half customer in basis and half in test) regardless of the number of entries each customer has without having to compute the exact split_frac or the cut_off value.

Reply:
So if we take the following case 
![image](https://user-images.githubusercontent.com/32060943/140818790-0b78cecb-e267-4d30-b2a5-658fcede4982.png)

Reply:
split_distinct will results in two dataframes of equal sizes wen applying ._split_dataframe
![image](https://user-images.githubusercontent.com/32060943/140818969-4a0ffba0-0848-4a25-b61c-cf6fa726acf1.png)

Reply:
While split_distinct = True results in two set of customer ID with equal size
![image](https://user-images.githubusercontent.com/32060943/140819127-c48c4344-bad2-49cc-ad68-1cc725aaf61b.png)

Reply:
@solegalli I guess we first need to agree on that behaviour and then I will add the test you suggested.

Reply:
I see also the comment on the docstring is not clear about that. Once we agreed on the behaviour, I'll adjust it.